scripts/linters/python_linter.py

```diff
@@ -417,34 +377,39 @@ def _lint_py_files(self, config_pylint, config_pycodestyle):
                 python_utils.PRINT('Linting Python files %s to %s...' % (
                     current_batch_start_index + 1, current_batch_end_index))
 
-            with linter_utils.redirect_stdout(stdout):
+            with linter_utils.redirect_stdout(sys.stdout):
                 # This line invokes Pylint and prints its output
                 # to the target stdout.
                 pylint_report = StringMessageStream()
                 pylinter = lint.Run(
                     current_files_to_lint + [config_pylint],
                     reporter=text.TextReporter(pylint_report),
                     exit=False).linter
+
+            with linter_utils.redirect_stdout(stdout):
                 # These lines invoke Pycodestyle and print its output
                 # to the target stdout.
                 style_guide = pycodestyle.StyleGuide(
                     config_file=config_pycodestyle)
                 pycodestyle_report = style_guide.check_files(
                     paths=current_files_to_lint)
 
-            if pylinter.msg_status != 0 or pycodestyle_report.get_count() != 0:
+            if pycodestyle_report.get_count() != 0:
                 summary_message = stdout.getvalue()
```

Comment:
> we were not appending result of pycodestyle to the messages, which is fixed now.

I was expecting pycodestyle errors to be in stdout earlier, can you explain why it wasn't getting added earlier?

Also, I feel think this was the reason behind the error message getting printed twice, can you check?

Reply:
Yes, this was the reason behind messages getting printed twice, this PR also fix that PR.

> I was expecting pycodestyle errors to be in stdout earlier, can you explain why it wasn't getting added earlier?

Last time it somehow doesn't catch my eye.

Reply:
> Yes, this was the reason behind messages getting printed twice, this PR also fix that PR.

Because of this, we had introduced pylint_reporter, right? SO now if we know what's was exactly causing this can we remove pylint_report?

> > I was expecting pycodestyle errors to be in stdout earlier, can you explain why it wasn't getting added earlier?

> Last time it somehow doesn't catch my eye.

So how can you say this resolves the issue! 

(Considering this PR is important for develop, I would suggest a quick meeting and explain to me the reason behind these changes!)

Reply:
We can meet quickly (lets have a chat about this).

Tbh, even I'm confused!